File: Back-End-M/src/main/java/com/attendance/fpt/services/StatisticService.java
package com.attendance.fpt.services;

import com.attendance.fpt.model.response.*;
import jakarta.servlet.http.HttpServletResponse;

import java.util.List;

public interface StatisticService {
    AttendanceWeeklyResponse getWeeklyAttendanceStatistics();

    List<LeaveOverallResponse> getLeaveOverallStatistics();

    long getTotalLeaveInDay();


    List<TopStaffAttendanceResponse> getTopFiveStaffAttendanceByMonth(int month,int year);

    List<OverallStatisticMonthResponse> getOverallByMonth(int month, int year);

    List<LeaveOverallResponse> getLeaveOverallStatisticsByMonth(int month, int year);

    void exportStatisticToExcel(int month, int year, HttpServletResponse response);

    List<OverallStatisticEmployeeYearResponse> getOverallAttendanceEmployeeByYear(int year);

    StatisticOverallEmployeeResponse getOverallEmployee( int month, int year);

    List<LeaveOverallResponse> getLeaveOverallEmployeeStatistics( int year);
}

File: Front-End/src/services/statistic.service.ts
import http from "@/utils/http";

export const statisticApi = {
  getWeeklyAttendanceStatistics: async () => {
    const response = await http.get("/statistics/weekly-attendance");
    return response.data;
  },
  getTotalLeaveInDay: async () => {
    const response = await http.get("/statistics/total-leave-in-day");
    return response.data;
  },
  getLeaveOverallStatistics: async () => {
    const response = await http.get("/statistics/leave-overall");
    return response.data;
  },

  getTopFiveStaffAttendanceByMonth: async (month: number, year: number) => {
    const response = await http.get(
      `/statistics/top-five-staff-attendance/${month}/${year}`
    );
    return response.data;
  },

  getOverallByMonth: async (month: number, year: number) => {
    const response = await http.get(`/statistics/overall/${month}/${year}`);
    return response.data;
  },

  getLeaveOverallStatisticsByMonth: async (month: number, year: number) => {
    const response = await http.get(
      `/statistics/leave-overall/${month}/${year}`
    );
    return response.data;
  },
  exportOverallByMonth: async (month: number, year: number) => {
    const response = await http.get(`/statistics/export/${month}/${year}`, {
      responseType: "blob",
    });
    return response.data;
  },

  getOverallAttendanceEmployeeByYear: async (year: number) => {
    const response = await http.get(`/statistics/overall-attendance/${year}`);
    return response.data.data;
  },
  getOverallEmployee: async (month: number, year: number) => {
    const response = await http.get(
      `/statistics/overall-employee/${month}/${year}`
    );
    return response.data.data;
  },
  getLeaveOverallEmployeeStatistics: async (year: number) => {
    const response = await http.get(
      `/statistics/leave-overall-employee/${year}`
    );
    return response.data.data;
  },
};
